packages/react-components/src/Dropdown.tsx

```diff
@@ -40,31 +40,37 @@ interface Props<Option> extends BareProps {
 }
 
 function Dropdown<Option> ({ allowAdd = false, className, defaultValue, dropdownClassName, help, isButton, isDisabled, isError, isMultiple, label, labelExtra, onAdd, onBlur, onChange, onClose, onSearch, options, placeholder, renderLabel, searchInput, style, transform, withEllipsis, withLabel, value }: Props<Option>): React.ReactElement<Props<Option>> {
+  const valueProp = useMemo(
+    () => isUndefined(value) ? defaultValue : value,
+    [defaultValue, value]
+  );
   const [stateValue, setStateValue] = useState<any>();
 
   const _onAdd = (_: React.SyntheticEvent<HTMLElement>, { value }: DropdownProps): void =>
     onAdd && onAdd(value);
 
-  const _onChange = (_: React.SyntheticEvent<HTMLElement> | null, { value }: DropdownProps): void => {
+  const handleOnChange = (value: any): void => {
     setStateValue(value);
-
     onChange && onChange(
       transform
         ? transform(value)
         : value
     );
   };
 
-  useEffect((): void => {
-    const newValue = isUndefined(value)
-      ? defaultValue
-      : value;
+  const _onChange = (_: React.SyntheticEvent<HTMLElement> | null, { value }: DropdownProps): void => {
+    if (JSON.stringify({ v: valueProp }) === JSON.stringify({ v: value })) {
+      return;
+    }
+    handleOnChange(value);
+  };
 
-    // only update parent if we have had something changed
-    if (JSON.stringify({ v: newValue }) !== JSON.stringify({ v: stateValue })) {
-      _onChange(null, { value: newValue });
+  useEffect(() => {
+    if (JSON.stringify({ v: valueProp }) === JSON.stringify({ v: stateValue })) {
```

Comment:
Not 100% sure on this one - since we use the stateValue internally, it may be “captured”. So if we don’t want to fire the effect of n it, and not need it on re-render it will probably make sense to move “stateValue” to as useRef.